filters/HAGDEMFilter.cpp

```diff
@@ -0,0 +1,130 @@
+/******************************************************************************
+* Copyright (c) 2020, Julian Fell (hi@jtfell.com)
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following
+* conditions are met:
+*
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include "HAGDEMFilter.hpp"
+
+#include <pdal/GDALUtils.hpp>
+
+namespace pdal
+{
+
+static StaticPluginInfo const s_info
+{
+    "filters.hag_dem",
+    "Computes height above ground using a DEM raster.",
+    "http://pdal.io/stages/filters.hag_dem.html"
+};
+
+CREATE_STATIC_STAGE(HAGDEMFilter, s_info)
+
+std::string HAGDEMFilter::getName() const
+{
+    return s_info.name;
+}
+
+
+HAGDEMFilter::HAGDEMFilter()
+{}
+
+
+void HAGDEMFilter::addArgs(ProgramArgs& args)
+{
+    args.add("raster", "GDAL-readable raster to use for DEM (uses band 1, "
```

Comment:
If you use `args.add(...).setPositional()`, you will this option will be required and you won't need to check that it was set in `prepared()`.